### PR TITLE
Add logic to account for cost increases.

### PR DESCRIPTION
**Pull request by willardmr**
Not great test coverage at this time and imperfect refactoring.  But hopefully the code is clear?

### CODE REVIEW

File: core/gcp/terraform_gcp_example_test.go [hunk 1]
```diff
@@ -61,37 +62,49 @@ func TestTerraformGcpExample(t *testing.T) {
...
+func getCostIncreasePercentage(currentRegion string, newRegion string) float64 {
+	cloudCosts := getCloudCosts()
+	return 100 * (cloudCosts[strings.ToLower(newRegion)].Cost - cloudCosts[strings.ToLower(currentRegion)].Cost) / cloudCosts[strings.ToLower(newRegion)].Cost
+}
+
+/* Best means lowest carbon intensity without too high of a price increase */
+func getBestRegion(currentRegion string, maximumCostIncreasePercentage float64) CloudIntensity {
+	cloudIntensities := getCloudIntensities()
+
+	lowestIntensity := 100000.0
```
Comment (AntoineSebert) [inline core/gcp/terraform_gcp_example_test.go:100]: How was this number chosen ?

File: core/gcp/impact.go [hunk 2]
```diff
@@ -9,18 +9,12 @@ import (
...
-	GeneralRegion  string
+	Provider      string
+	Region        string
+	Location      string
+	Impact        float64
+	Source        string
+	GeneralRegion string
 }
 
 func loadCloudIntensities(data [][]string) []CloudIntensity {
```
Comment (AntoineSebert) [inline core/gcp/impact.go:20]: This function could be taken out of the example file, and be place in a dedicated module, `import` maybe ?

File: core/gcp/terraform_gcp_example_test.go [hunk 3]
```diff
@@ -61,37 +62,49 @@ func TestTerraformGcpExample(t *testing.T) {
...
-						bestRegion = intensity
-						lowestIntensity = intensity.Impact
-					}
-				}
+				bestRegion := getBestRegion(region, maximumCostIncreasePercentage)
 				if strings.ToLower(bestRegion.Region) != strings.ToLower(region) {
 					assert := assert.New(t)
-					assert.Equal(bestRegion.Region, region, fmt.Sprintf("Resource of type %s with name %s is being created in region %s, but it should be created in region %s to reduce carbon emissions.", resourceChanges.Type, plannedValues.AttributeValues["name"].(string), strings.ToLower(region), bestRegion.Region))
+					assert.Equal(bestRegion.Region, region,
+						fmt.Sprintf("Resource of type %s with name %s is being created in region %s, but it should be created in region %s to reduce carbon emissions.", resourceChanges.Type, plannedValues.AttributeValues["name"].(string), strings.ToLower(region), bestRegion.Region))
```
Comment (AntoineSebert) [inline core/gcp/terraform_gcp_example_test.go:84]: The user might also be interested in seeing the previous carbon emissions vs the new carbon emissions

Comment (willardmr) [reply core/gcp/terraform_gcp_example_test.go:84]: Sure, maybe you can create an issue to add that feature?

File: core/gcp/impact.go (hunk 2 above)
Comment (willardmr) [reply core/gcp/impact.go:20]: Can you elaborate?  What makes this an "example file"?

**Pull request merged**